 Settings:Bugfix for NPE, Only valid screen power consumption need to smear
 
diff --git a/core/java/com/android/internal/os/BatteryStatsHelper.java b/core/java/com/android/internal/os/BatteryStatsHelper.java
index 83880e1..8256a8d 100644
--- a/core/java/com/android/internal/os/BatteryStatsHelper.java
+++ b/core/java/com/android/internal/os/BatteryStatsHelper.java
@@ -854,7 +854,10 @@ public double removeHiddenBatterySippers(List<BatterySipper> sippers) {
             }
         }
 
-        smearScreenBatterySipper(sippers, screenSipper);
+        //Only valid screen power consumption need to smear
+        if(screenSipper != null){
+           smearScreenBatterySipper(sippers, screenSipper);
+        }
 
         return proportionalSmearPowerMah;
     }
